bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MessageBox.java

```diff
@@ -101,6 +107,68 @@ public MessageBox (Shell parent, int style) {
 	checkSubclass ();
 }
 
+long CBTProc(long nCode, long wParam, long lParam) {
+	if (labels != null && (int) nCode == OS.HCBT_ACTIVATE) {
+		/* The system is about to activate MessageBox window */
+		switch (buttonBits) {
+		case OS.MB_OKCANCEL:
+			setButtonText(wParam, SWT.CANCEL, OS.IDCANCEL);
+		case OS.MB_OK:
+			setButtonText(wParam, SWT.OK, OS.IDOK);
+			break;
+		case OS.MB_YESNOCANCEL:
+			setButtonText(wParam, SWT.CANCEL, OS.IDCANCEL);
+		case OS.MB_YESNO:
+			setButtonText(wParam, SWT.YES, OS.IDYES);
+			setButtonText(wParam, SWT.NO, OS.IDNO);
+			break;
+		case OS.MB_RETRYCANCEL:
+			setButtonText(wParam, SWT.RETRY, OS.IDRETRY);
+			setButtonText(wParam, SWT.CANCEL, OS.IDCANCEL);
+			break;
+		case OS.MB_ABORTRETRYIGNORE:
+			setButtonText(wParam, SWT.ABORT, OS.IDABORT);
+			setButtonText(wParam, SWT.RETRY, OS.IDRETRY);
+			setButtonText(wParam, SWT.IGNORE, OS.IDIGNORE);
+			break;
+		default:
+			break;
+		}
+		/* Exit the CBT hook */
+		OS.UnhookWindowsHookEx(cbtHook);
```

Comment:
I understand that you copied the comment from the example, but it's not very useful. I'd just delete it.

The bigger problem here is that you call `OS.CallNextHookEx()` on an already destroyed hook instead of just returning. Note that the example had a different problem - it didn't call the next hook, which may be present. The proper approach would be to call next hook and THEN uninstall and return.

Reply:
Done.